lib/logstash/dead_letter_post_office.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+
+class LogStash::DeadLetterPostOffice
+
+  def self.logger=(logger)
+    @logger = logger
+  end
+
+  def self.logger
+    @logger ||= Cabin::Channel.get(LogStash)
+  end
+
+  def self.destination=(destination)
+    logger.info("Dead letter events will be sent to \"#{destination.location}\".")
+    @destination = destination
+  end
+
+  def self.<<(events)
+    events = [events] unless events.is_a?(Array)
+
+    events.each do |event|
+      logger.warn("dead letter received!", :event => event.to_hash)
+      event.tag("_dead_letter")
+      event.cancel
+      @destination << event
+    end
+  end
+
+  module Destination
+
+    class Base
+      def location; end
+      def <<(event); end
```

Comment:
same concern for << here, kinda, though the semantics seem different

`DeadLetterPostOffice#<<` will accept an array and do `#each ... @destination << object ...` but this one takes a single object and does not iterate over it.

So two things, I suppose:
1) Different parameter handling of the class method `#<<` and the `Destination::Base#<<` feels weird
2) The method `<<` feels like the wrong one to use.

Thoughts?